### PR TITLE
Add ability to specify image name to Cloud Plugin

### PR DESCRIPTION
**Pull request by IndushaS**
Summary
Add the ability to the Plugin for the user to specify the image name for create and upgrade command.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-32327

#### Release Note

```release-note
Add ability to specify image name to Cloud Plugin
```

### CODE REVIEW

File: server/command_create.go [hunk 1]
```diff
@@ -64,6 +65,15 @@ func parseCreateArgs(args []string, install *Installation) error {
 		return errors.Errorf("invalid license option %s, must be %s, %s or %s", install.License, licenseOptionE10, licenseOptionE20, licenseOptionTE)
 	}
 
+	install.Image, err = createFlagSet.GetString("image")
+	if err != nil {
+		return err
+	}
+
+	if !validImageName(install.Image) {
+		return errors.Errorf("invalid image name %s", strings.Join(dockerRepoWhitelist, ", "))
```
Comment (gabrieljackson) [inline server/command_create.go:74]:
```suggestion
		return errors.Errorf("invalid image name %s, valid options are %s", install.Image, strings.Join(dockerRepoWhitelist, ", "))
```

File: server/command_upgrade.go [hunk 2]
```diff
@@ -41,13 +42,19 @@ func buildPatchInstallationRequestFromArgs(args []string) (*cloud.PatchInstallat
...
+		return nil, err
+	}
+	if version == "" && license == "" && size == "" && image == "" {
+		return nil, errors.New("must specify at least one option: version, license, image or size")
 	}
 	if license != "" && !validLicenseOption(license) {
 		return nil, errors.Errorf("invalid license option %s; must be %s, %s or %s", license, licenseOptionE10, licenseOptionE20, licenseOptionTE)
 	}
+	if image != "" && !validImageName(image) {
+		return nil, errors.Errorf("invalid image name %s", strings.Join(dockerRepoWhitelist, ", "))
```
Comment (gabrieljackson) [inline server/command_upgrade.go:56]: Same thing here.

**Review by gabrieljackson — CHANGES_REQUESTED** [review]
Awesome, just two small issues to address.

File: server/command_upgrade.go (hunk 2 above)
Comment (IndushaS) [reply server/command_upgrade.go:56]: Thanks Gabe, I'll go ahead and fix that

**Review by gabrieljackson — APPROVED** [review]
👍

File: server/command_create.go [hunk 3]
```diff
@@ -12,6 +12,7 @@ import (
 	flag "github.com/spf13/pflag"
 )
 
+var dockerRepoWhitelist = []string{"mattermost/mattermost-enterprise-edition", "mattermost/mm-ee-test", "mattermost/mm-ee-cloud", "mattermost/mm-te", "mattermost/mattermost-team-edition"}
```
Comment (gigawhitlocks) [inline server/command_create.go:15]: Just a style request: could we put newlines after each of these items in this list just so the list appears vertically?

**Review by gigawhitlocks — APPROVED** [review]
LGTM, just one tiny style nitpick/request

File: server/command_create.go (hunk 3 above)
Comment (IndushaS) [reply server/command_create.go:15]: I'll go ahead and do that Ian thank you for the feedback

**Pull request merged**